src/od/od.rs

```diff
@@ -46,446 +104,345 @@ pub fn uumain(args: Vec<String>) -> i32 {
-    let stdnionly = [InputSource::Stdin];
-    let inputs = args[1..]
-        .iter()
-        .filter_map(|w| match w as &str {
-            "--" => Some(InputSource::Stdin),
-            o if o.starts_with("-") => None,
-            x => Some(InputSource::FileName(x)),
-        })
-        .collect::<Vec<_>>();
-    // If no input files named, use stdin.
-    let inputs = if inputs.len() == 0 {
-        &stdnionly[..]
-    } else {
-        &inputs[..]
-    };
-    // Gather up format flags, we don't use getopts becase we need keep them in order.
-    let flags = args[1..]
-        .iter()
-        .filter_map(|w| match w as &str {
-            "--" => None,
-            o if o.starts_with("-") => Some(&o[1..]),
-            _ => None,
-        })
-        .collect::<Vec<_>>();
+struct OdOptions {
+    byte_order: ByteOrder,
+    skip_bytes : usize,
+    read_bytes : Option<usize>,
+    label: Option<usize>,
+    input_strings: Vec<String>,
+    formats: Vec<ParsedFormatterItemInfo>,
+    line_bytes: usize,
+    output_duplicates: bool,
+    radix: Radix,
+}
 
-        // At the moment, char (-a & -c)formats need the driver to set up a
-        // line by inserting a different # of of spaces at the start.
-        struct OdFormater {
-            writer: fn(p: u64, itembytes: usize),
-            offmarg: usize,
-        };
-        let oct = OdFormater {
-            writer: print_item_oct,  offmarg: 2
-        };
-        let hex = OdFormater {
-            writer: print_item_hex, offmarg: 2
-        };
-        let dec_u = OdFormater {
-            writer: print_item_dec_u, offmarg: 2
+impl OdOptions {
+    fn new(matches: getopts::Matches, args: Vec<String>) -> Result<OdOptions, String> {
+        let byte_order = match matches.opt_str("endian").as_ref().map(String::as_ref) {
+            None => { ByteOrder::Native },
+            Some("little") => { ByteOrder::Little },
+            Some("big") => { ByteOrder::Big },
+            Some(s) => {
+                return Err(format!("Invalid argument --endian={}", s));
+            }
         };
-        let dec_s = OdFormater {
-            writer: print_item_dec_s, offmarg: 2
+
+        let mut skip_bytes = match matches.opt_default("skip-bytes", "0") {
+            None => 0,
+            Some(s) => {
+                match parse_number_of_bytes(&s) {
+                    Ok(i) => { i }
+                    Err(_) => {
+                        return Err(format!("Invalid argument --skip-bytes={}", s));
+                    }
+                }
+            }
         };
-        let a_char = OdFormater {
-            writer: print_item_a, offmarg: 1
+
+        let mut label: Option<usize> = None;
+
+        let input_strings = match parse_inputs(&matches) {
+            Ok(CommandLineInputs::FileNames(v)) => v,
+            Ok(CommandLineInputs::FileAndOffset((f, s, l))) => {
+                skip_bytes = s;
+                label = l;
+                vec!{f}
```

Comment:
I don't see people use curly braces for `vec!` very often.  Usually they use square brackets or parentheses.